src/Microsoft.AspNetCore.Razor.Evolution/RazorTemplateEngineResult.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public struct RazorTemplateEngineResult
```

Comment:
Rather than make things awkward, the intent is provide a set of flexible methods that work with any of the types you might want.